acceptance/openstack/rds/v3/rds_test.go

```diff
@@ -56,89 +45,335 @@ func TestRdsLifecycle(t *testing.T) {
 	th.AssertNoErr(t, err)
 
-	cc, err := clients.CloudAndClient()
+	t.Log("Restart")
+
+	restart, err := instances.Restart(client, instances.RestartOpts{InstanceId: rds.Id, Restart: struct{}{}})
+	th.AssertNoErr(t, err)
+	err = instances.WaitForJobCompleted(client, 600, *restart)
 	th.AssertNoErr(t, err)
 
-	// Create RDSv3 instance
-	rds := createRDS(t, client, cc.RegionName)
-	defer deleteRDS(t, client, rds.Id)
-	th.AssertEquals(t, rds.Volume.Size, 100)
+	if err := instances.WaitForStateAvailable(client, 600, rds.Id); err != nil {
+		t.Fatalf("Status available wasn't present")
+	}
+
+	t.Log("StopInstance")
+
+	stop, err := instances.StopInstance(client, rds.Id)
+	th.AssertNoErr(t, err)
+	err = instances.WaitForJobCompleted(client, 600, *stop)
+	th.AssertNoErr(t, err)
+
+	time.Sleep(5 * time.Second)
+
+	t.Log("StartupInstance")
+
+	start, err := instances.StartupInstance(client, rds.Id)
+	th.AssertNoErr(t, err)
+	err = instances.WaitForJobCompleted(client, 600, *start)
+	th.AssertNoErr(t, err)
+
+	t.Log("Resize")
+
+	if err := instances.WaitForStateAvailable(client, 600, rds.Id); err != nil {
+		t.Fatalf("Status available wasn't present")
+	}
+
+	resize, err := instances.Resize(client, instances.ResizeOpts{
+		InstanceId: rds.Id,
+		SpecCode:   "rds.pg.c2.large",
+	})
+	th.AssertNoErr(t, err)
+	err = instances.WaitForJobCompleted(client, 600, *resize)
+	th.AssertNoErr(t, err)
 
 	t.Logf("Attempting to create RDSv3 Read Replica")
 
-	prefix := "rds-rr-"
-	rdsReplicaName := tools.RandomString(prefix, 8)
+	rdsReplicaName := tools.RandomString("rds-rr-", 8)
 	kmsID := clients.EnvOS.GetEnv("KMS_ID")
 	az := clients.EnvOS.GetEnv("AVAILABILITY_ZONE")
 	if az == "" {
 		az = "eu-de-01"
 	}
 
-	createOpts := instances.CreateReplicaOpts{
+	if err := instances.WaitForStateAvailable(client, 600, rds.Id); err != nil {
+		t.Fatalf("Status available wasn't present")
+	}
+
+	replica, err := instances.CreateReplica(client, instances.CreateReplicaOpts{
 		Name:             rdsReplicaName,
 		ReplicaOfId:      rds.Id,
 		DiskEncryptionId: kmsID,
-		FlavorRef:        "rds.pg.c2.medium.rr",
+		FlavorRef:        "rds.pg.c2.large.rr",
 		Volume: &instances.Volume{
 			Type: "COMMON",
 			Size: 100,
 		},
 		AvailabilityZone: az,
-	}
+	})
+	th.AssertNoErr(t, err)
+	err = instances.WaitForJobCompleted(client, 1200, replica.JobId)
+	th.AssertNoErr(t, err)
+
+	t.Logf("Created RDSv3 Read Replica: %s", replica.Instance.Id)
+
+	t.Cleanup(func() {
+		t.Logf("Attempting to delete RDSv3 Read Replica: %s", replica.Instance.Id)
+		_, err := instances.Delete(client, replica.Instance.Id)
+		th.AssertNoErr(t, err)
+		t.Logf("RDSv3 Read Replica instance deleted: %s", replica.Instance.Id)
+	})
 
-	createResult := instances.CreateReplica(client, createOpts)
-	rdsReadReplica, err := createResult.Extract()
+	t.Log("UpdateDataIp")
+
+	ip, err := security.UpdateDataIp(client, security.UpdateDataIpOpts{
+		InstanceId: rds.Id,
+		NewIp:      "192.168.30.254",
```

Comment:
it possible to obtain this ip from subnet cidr + random?

Reply:
It's basically a random private IP, so it's not very important, and we don't have a list of available private IPs (only a list of used) API.